test/export/InstanceExporter.test.ts

```diff
@@ -924,6 +924,64 @@ describe('InstanceExporter', () => {
       });
     });
 
+    it('should fix a reference to a type based on a profile', () => {
+      const profiledObservation = new Profile('USCoreObservation');
+      profiledObservation.parent =
+        'http://hl7.org/fhir/us/core/StructureDefinition/us-core-observation-lab';
+      doc.profiles.set(profiledObservation.name, profiledObservation);
+
+      const profiledInstance = new Instance('MyExampleObservation');
+      profiledInstance.instanceOf = 'USCoreObservation';
+      const fixedRefRule = new FixedValueRule('subject');
+      fixedRefRule.fixedValue = new FshReference('Bar');
+      profiledInstance.rules.push(fixedRefRule);
+      doc.instances.set(profiledInstance.name, profiledInstance);
+      doc.instances.set(patientInstance.name, patientInstance);
+      const exported = exportInstance(profiledInstance);
+      expect(exported.subject).toEqual({
+        reference: 'Patient/Bar'
+      });
+    });
```

Comment:
I am not sure I understand what this test is up to, but that could definitely just be me not understanding. So could you give an example of what this would look like in FSH?

Reply:
Ok wait I think I've figured it out. But two things I don't understand. One, can't we just directly set the `instanceOf` on the `profiledInstance` to be `http://hl7.org/fhir/us/core/StructureDefinition/us-core-observation-lab`? Instead of creating `profiledObservation`. And two, what is the function of `doc.instances.set(patientInstance.name, patientInstance)`?

Reply:
You're right, I'll switch to just directly set the `instanceOf` and I think the `patientInstance` line was a copy and paste mistake so I'll remove it. I'll also see if I can maybe add some comments to clarify what the test is doing to make it easier to understand.